### DISCUSSION TITLE
Cannot load the CustomBA in WiX 4, Error 0x80131902

### DISCUSSION THREAD

**Discussion by MatthiasRInst** [wixtoolset/issues#8603] [Questions]
Hello,

having migrated my CustomBA from WiX 3.* to 4.0.5, it builds, but Burn does not call it.

I think I have considered all the necessary steps to bind the CustomBA to the bundle:

Bundle.wxs:

```
  <Bundle>
    <BootstrapperApplication>
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\ManagedBootstrapper.CustomBA.dll"
               bal:BAFactoryAssembly="yes" />
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\mbanative.dll" />
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\WixToolset.Mba.Core.dll" />
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\WixToolset.Dtf.WindowsInstaller.dll" />
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\WixToolset.Mba.Host.config" />
      <bal:WixManagedBootstrapperApplicationHost />
    </BootstrapperApplication>

```

.NET Framework is contained as an ExePackage with bal:PrereqPackage="yes".

In my CustomBA (assembly is ManagedBootstrapper.CustomBA.dll), there is a factory class derived from BaseBootstrapperApplicationFactory where the Create method just creates a new CustomBootstrapperApplication object and sets its Command property to the command parameter.

My CustomBootstrapperApplication class is derived from BootstrapperApplication.

In the AssemblyInfo.cs, there is the attribute `[assembly: BootstrapperApplicationFactory(typeof(CustomBootstrapperApplicationFactory))] 
` instead of the old BootstrapperApplication attribute.

I created WixToolset.Mba.Host.config from the old BootstrapperCore.config:

```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
  <configSections>
    <sectionGroup name="wix.bootstrapper" type="WixToolset.MbaCore.BootstrapperSectionGroup, Wix.Mba.Host">
      <section name="host" type="WixToolset.MbaCore.HostSection, Wix.Mba.Host" />
    </sectionGroup>
  </configSections>
  <startup useLegacyV2RuntimeActivationPolicy="true">
    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.8" />
  </startup>

  <wix.bootstrapper>
    <host assemblyName="ManagedBootstrapper.CustomBA">
    </host>
  </wix.bootstrapper>
</configuration>

```

CustomBA and Bundle build, but when I run my Install.exe, log file tells me:

```
[55B4:18EC][2024-06-28T19:19:33]i000: Loading managed bootstrapper application.
[55B4:18EC][2024-06-28T19:19:33]e000: Error 0x80131902: Failed to create the bootstrapper application.
[55B4:18EC][2024-06-28T19:19:33]e000: Error 0x80131902: Failed to create the managed bootstrapper application.
[55B4:18EC][2024-06-28T19:19:33]e000: Error 0x80131902: Failed to create BA.
[55B4:18EC][2024-06-28T19:19:33]e000: Error 0x80131902: Failed to load BA.
[55B4:18EC][2024-06-28T19:19:33]e000: Error 0x80131902: Failed while running 
```

I have written an Exe wrapper that calls CustomBootstrapperApplicationFactory.Create() and then CustomBootstrapperApplication.Run(). Using this, CustomBA starts.

As of now, I have no clue what is missing here.

**Comment by gaja2015** [comment]
I am also getting the same issue

**Comment by gaja2015** [comment]
It will be very helpful, if implementer of this V4 Wix group to provide a sample working project(Hello World Application)  for CustomBootStrapper application in V4.

**Comment by michaelcourtin-seequent** [comment]
Did either of you ever figure out what the problem was? I too am migrating from Wix 3 to 4 and encountering exactly the same issue.

**Comment by devikandixit** [comment]
Hi @michaelcourtin-seequent , I am able to fix this issue now.
Below  things are necessary as mentioned by  [MatthiasRInst](https://github.com/MatthiasRInst)
`  <Bundle>
    <BootstrapperApplication>
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\ManagedBootstrapper.CustomBA.dll"
               bal:BAFactoryAssembly="yes" />
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\mbanative.dll" />
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\WixToolset.Mba.Core.dll" />
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\WixToolset.Dtf.WindowsInstaller.dll" />
      <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\WixToolset.Mba.Host.config" />
      <bal:WixManagedBootstrapperApplicationHost />
    </BootstrapperApplication>
`
Above needs to be added in Bundle.wxs file.

    Apart from that there is one more thing.
  Here <Payload SourceFile="$(var.BranchOutput)\CustomBA\bin\$(var.Configuration)\ManagedBootstrapper.CustomBA.dll"
  
  ManagedBootstrapper.CustomBA.dll is a main Bootstrapper Application.  This application (mine is a C# code) also needs to refer the 
  **WixToolset.Mba.Core.dll**
  I added WixToolset.Mba.Core.dll as a reference . This fixed my issue.
  
  Errors what we are getting is basically library is not loaded (fail to load WixToolset.Mba.Core.dll)failure.
  
  Please try to include WixToolset.Mba.Core.dll as the reference and test.
